server/src/api/v1.py

```diff
@@ -157,8 +157,31 @@ def check_token_queue(auth_token: str, secret_key: str, queue: str) -> bool:
     return queue in allowed_queues
 
 
+def check_token_reservation_timeout(
+    auth_token: str, secret_key: str, reservation_timeout: int, queue: str
+) -> bool:
+    """
+    Checks if the requested reservation is either less than the max
+    or that their token gives them the permission to use a higher one
+    """
+    # Max reservation time defaults to 6 hours
+    max_reservation_time = 6 * 60 * 60
+    if reservation_timeout <= max_reservation_time:
+        return True
+    decoded_jwt = decode_jwt_token(auth_token, secret_key)
+    max_reservation_time_dict = decoded_jwt.get("max_reservation_time", {})
+    queue_reservation_time = max_reservation_time_dict.get(queue, 0)
+    star_reservation_time = max_reservation_time_dict.get("*", 0)
+    max_reservation_time = max(queue_reservation_time, star_reservation_time)
+    return reservation_timeout <= max_reservation_time
+
+
 def check_token_permissions(
-    auth_token: str, secret_key: str, priority: int, queue: str
+    auth_token: str,
+    secret_key: str,
+    priority: int,
+    queue: str,
+    reservation_timeout: int,
```

Comment:
Every time you add a new permissions check you will probably need to add more parameters here. I think these three should be replaced by the entire job dict (I think it's called `data` in the `job_builder` method where this is called). What this method is doing (conceptually) is to check whether the permissions specified in the `auth_token` allow this job to be submitted. You can add as many permission checks as you like, the method signature would still be the same.